### PR TITLE
Feature/san 5258 balance view

### PR DESCRIPTION
**Pull request by GarageInc**
Done:
* implemented SAN-5258 logic for "held assets by address" for balance view page and trigger form
image
image
image
image

* new settings for chart in balance view page
image
image

### CODE REVIEW

**Comment by haritonasty** [comment]
looks like you use wrong `Select`. As I see, this component not in `san-ui`, it's [here](https://github.com/santiment/app/blob/3c64ff06e5a78e5d9221d8fe5a58253ed63b3b18/src/components/Select/Select.js)

is it possible to change?

**Comment by GarageInc** [comment]
@haritonasty , I found such comment from Vanguard
image
"TODO(vanguard): change to the san-ui dropdown select" (c)

And changed to using san-ui select for new components. (without  LibSelect from 'react-virtualized-select' in src/components/Select/Select.js)

File: src/ducks/HistoricalBalance/GetHistoricalBalance.js [hunk 1]
```diff
@@ -59,22 +65,22 @@ class GetHistoricalBalance extends Component {
...
           variables: {
             slug,
-            address: this.props.wallet,
-            interval: '1d',
-            to: new Date().toISOString(),
-            from: '2017-12-01T16:28:22.486Z'
+            address: wallet,
+            interval: interval,
+            to: to,
+            from: from
```
Comment (haritonasty) [inline src/ducks/HistoricalBalance/GetHistoricalBalance.js]: You can write it shortly
`to: to,` -> `to,` and other with the same way

File: src/ducks/HistoricalBalance/balanceView/BalanceView.js [hunk 2]
```diff
@@ -1,125 +1,186 @@
...
   }
 
-  handleAssetsChange = assets => {
-    this.setState({ assets: assets.map(asset => asset.value) }, () => {
-      this.props.onChangeQuery(this.state)
+  const onCalendarChange = ([from, to]) => {
+    setChartSettings({
+      ...chartSettings,
+      ...{
+        from: from.toISOString(),
```
Comment (haritonasty) [inline src/ducks/HistoricalBalance/balanceView/BalanceView.js]: LOC 70 - `...` for what?

File: src/ducks/HistoricalBalance/AssetsField.js [hunk 3]
```diff
@@ -1,73 +1,84 @@
...
+    props: mapAssetsHeldByAddressToProps,
+    skip: ({ byAddress }) => {
+      return !byAddress
+    },
+    options: ({ byAddress }) => {
+      return {
+        variables: {
+          address: byAddress
+        },
+        fetchPolicy: 'cache-first'
```
Comment (DmitriVanGuard) [inline src/ducks/HistoricalBalance/AssetsField.js]: It's by default `cache-first`

File: src/ducks/HistoricalBalance/GetHistoricalBalance.js [hunk 4]
```diff
@@ -48,7 +45,16 @@ class GetHistoricalBalance extends Component {
   }
 
   fetchHistoricalBalance () {
-    this.props.assets.forEach(slug => {
+    const {
+      assets,
+      wallet,
+      client,
+      interval = '1d',
+      to = new Date().toISOString(),
```
Comment (DmitriVanGuard) [inline src/ducks/HistoricalBalance/GetHistoricalBalance.js]: Very cache "unfriendly". Either set it to the day/hour end, or use `getTimeIntervalFromToday`

File: src/ducks/HistoricalBalance/GetHistoricalBalance.js [hunk 5]
```diff
@@ -48,7 +45,16 @@ class GetHistoricalBalance extends Component {
...
+      from = '2017-12-01T16:28:22.486Z'
```
Comment (DmitriVanGuard) [inline src/ducks/HistoricalBalance/GetHistoricalBalance.js]: `'2017-12-01T16:28:22.486Z'` should be moved out as constant

File: src/ducks/HistoricalBalance/balanceView/BalanceChartHeader.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+
```
Comment (DmitriVanGuard) [inline src/ducks/HistoricalBalance/balanceView/BalanceChartHeader.js]: Empty line. Does not look cool 😎

File: src/ducks/HistoricalBalance/balanceView/BalanceView.js (hunk 2 above)
Comment (DmitriVanGuard) [reply src/ducks/HistoricalBalance/balanceView/BalanceView.js]: Same for `LOC 63`

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 7]
```diff
@@ -119,18 +148,42 @@ const enhance = compose(
...
     name: 'Projects',
     props: mapDataToProps,
+    skip: ({ byAddress }) => {
+      return !!byAddress
+    },
     options: () => {
       return {
-        errorPolicy: 'all'
+        errorPolicy: 'all',
+        fetchPolicy: 'cache-first'
```
Comment (DmitriVanGuard) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: `[D-1-2]` `cache-first` is a default policy

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 8]
```diff
@@ -119,18 +148,42 @@ const enhance = compose(
   graphql(pickGQL('all'), {
     name: 'Projects',
     props: mapDataToProps,
+    skip: ({ byAddress }) => {
+      return !!byAddress
+    },
```
Comment (DmitriVanGuard) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: `[D-1-1]` Could be written with less syntax noise: `skip: ({byAddress}) => byAddress`.

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 9]
```diff
@@ -119,18 +148,42 @@ const enhance = compose(
...
-        errorPolicy: 'all'
+        errorPolicy: 'all',
+        fetchPolicy: 'cache-first'
       }
     }
   }),
   graphql(pickGQL('erc20'), {
     name: 'Projects',
     props: mapDataToProps,
+    skip: ({ byAddress }) => {
```
Comment (DmitriVanGuard) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: Same as `[D-1-1]`

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js [hunk 10]
```diff
@@ -119,18 +148,42 @@ const enhance = compose(
...
+      return !!byAddress
+    },
     options: () => {
       return {
-        errorPolicy: 'all'
+        errorPolicy: 'all',
+        fetchPolicy: 'cache-first'
```
Comment (DmitriVanGuard) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormAssetWallet.js]: Same as `[D-1-2]`

**Pull request merged**